.pre-commit-config.yaml

```diff
@@ -27,27 +27,27 @@ repos:
       - id: mixed-line-ending
         args: [--fix=lf]
   - repo: https://github.com/asottile/pyupgrade
-    rev: 'v3.3.1'
+    rev: 'v3.17.0'
     hooks:
       - id: pyupgrade
         args:
-          - --py38-plus
+          - --py310-plus
```

Comment:
This is the key change in this PR.

This, for example, led `pyupgrade` to rewrite some type hints using `typing.Union` to instead use `|`.